lib/note-detail.jsx

```diff
@@ -34,22 +30,34 @@ export default React.createClass( {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function( { note } ) {
-		this.setState( {
-			content: get( note, 'data.content', '' )
-		} );
+	componentWillReceiveProps: function() {
+		const { note } = this.props;
+
+		if ( ! this.isValidNote( note ) || ! this.noteEditor ) return;
+
+		if ( saveTimerId ) clearTimeout( saveTimerId );
+
+		// Save note if content changes haven't been saved yet
+		if ( note.data.content !== this.noteEditor.value ) {
+			this.props.onChangeContent( note, this.noteEditor.value );
+			analytics.tracks.recordEvent( 'editor_note_edited' );
+		}
 	},
 
 	componentDidUpdate: function() {
-		// Let's focus the editor for new and empty notes
 		const { note } = this.props;
-		const { content } = this.state;
-		if ( this.isValidNote( note ) && content === '' ) {
-			this.noteEditor.focus();
+		const content = get( note, 'data.content', '' );
+		if ( this.isValidNote( note ) && this.noteEditor ) {
+			this.noteEditor.value = content;
+
+			// Let's focus the editor for new and empty notes
+			if ( content === '' ) {
+				this.noteEditor.focus();
+			}
 		}
 	},
```

Comment:
I'm not sure if you are aware, but `componentDidUpdate( prevProps, prevState )` actually has access to the old props. It looks like we could accomplish in one spot what we are doing in the two lifecycle methods here.

If we use something like the debounce function from **lodash**, this would be a good spot to call its `flush` method.

```js
componentDidUpdate() {
	this.onEditorChanged.flush();

	…
}
```

that `flush()` method won't do anything if no changes are pending. In other words, if I haven't changed the note, then I haven't requested the update, so when `flush()` gets called [it has nothing to run](https://github.com/lodash/lodash/blob/4.6.1/lodash.js#L8824) and just returns. The only **catch** I see is that `this.saveNote()` relies on `this.props`, which when called from `componentDidUpdate()` are the new ones. I'm not sure if that matters or not.